ignite/distributed/comp_models/base.py

```diff
@@ -135,11 +125,26 @@ def _all_collective_op(
         tensor = fn(tensor, *args, **kwargs)
 
         if out_dtype is not None and tensor_device is not None:
-            tensor = tensor.to(dtype=out_dtype, device=tensor_device)
-        elif out_dtype is not None:
-            tensor = tensor.to(dtype=out_dtype)
-        elif tensor_device is not None:
-            tensor = tensor.to(device=tensor_device)
+            return tensor.to(dtype=out_dtype, device=tensor_device)
```

Comment:
Does it miss a test for the coverage below ?
But line 139 lets me think that device can't be None.

Reply:
This can be tested with a collective op applied on a CPU tensor for example but we are using CUDA backend etc.
I can add this.

Reply:
Actually, this case can not be tested and report coverage with our CI (needed XLA multi-TPUs)